hs_core/models.py

```diff
@@ -2679,10 +2679,15 @@ def readme_file(self):
         res_files_at_root = self.files.filter(file_folder='')
         readme_txt_file = None
         readme_md_file = None
+
+        def get_file_name(f):
+            return os.path.basename(f.get_storage_path(resource=self)).lower()
```

Comment:
@pkdash is this `get_file_name` func intended to reduce the number of queries by avoiding `f.file_name`?

Reply:
@devincowan That is correct. `f.file_name` results in one query per file since the logic for the `file_name` property access the `resource` property of the file which returns the `GenericForeignKey` field. Accessing `GenericForeignKey` results in a separate query. Since the resource for all the files of a resource is the same object, there is no need to access it for each file.